### PR TITLE
feat: add access_token_type argument in machine user (recreated)

### LINKED ISSUE #60: Add `access_token_type` argument in machine user
## Description

Since JWT access token type is available by the release of zitadel/zitadel#5150, we want to set up this parameter with Terraform.

### PR DESCRIPTION
**Pull request by TakumiHaruta**
## Description

Close #60
Close #61

- Add `access_token_type` argument in machine user

### CODE REVIEW

**Comment by TakumiHaruta** [comment]
@stebenz Hi, could you check and merge this PR? thanks

File: zitadel/v2/machine_user/resource.go [hunk 1]
```diff
@@ -51,6 +51,11 @@ func GetResource() *schema.Resource {
 				Optional:    true,
 				Description: "Description of the user",
 			},
+			accessTokenTypeVar: {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Access token type of the user",
```
[the remark this answers is not in the document]
Comment (TakumiHaruta) [reply zitadel/v2/machine_user/resource.go]: @stebenz 
There is no `validateFunc` used in other functions. I'm not sure my implementation meets your expectation but I just added it. Please have a look

[the remark this answers is not in the document]
Comment (stebenz) [reply zitadel/v2/machine_user/resource.go]: Totally missed telling you this, but there are already helper functions in place that you can use, if you have a look at the application_oidc and for example the authMethod or appType I hope it should be self-explanatory.
With the helper functions the ENUM will get directly used from the generated ZITADEL API.

[the remark this answers is not in the document]
Comment (TakumiHaruta) [reply zitadel/v2/machine_user/resource.go]: @stebenz Please check that again

**Review by stebenz — APPROVED** [review]

File: zitadel/v2/machine_user/resource.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (stebenz) [reply zitadel/v2/machine_user/resource.go]: Perfect

**Pull request merged**